lib/honeybadger/insights/absinthe.ex

```diff
@@ -0,0 +1,99 @@
+defmodule Honeybadger.Insights.Absinthe do
+  @moduledoc false
+
+  use Honeybadger.Insights.Base
+
+  @required_dependencies [Absinthe]
+
+  @telemetry_events [
+    "absinthe.execute.operation.stop",
+    "absinthe.execute.operation.exception"
+  ]
+
+  # This is not loaded by default since it can add a ton of events, but is here
+  # in case it is added to the insights_config.
+  def extract_metadata(%{resolution: resolution}, "absinthe.resolve.field.stop") do
+    %{
+      field_name: resolution.definition.name,
+      parent_type: resolution.parent_type.name,
+      state: resolution.state
+    }
+  end
+
+  def extract_metadata(meta, _name) do
+    %{
+      operation_name: get_operation_name(meta),
+      operation_type: get_operation_type(meta),
+      selections: get_graphql_selections(meta),
+      schema: get_schema(meta),
+      errors: get_errors(meta)
+    }
+  end
+
+  defp get_schema(%{blueprint: blueprint}) when is_map(blueprint), do: Map.get(blueprint, :schema)
+  defp get_schema(_), do: nil
+
+  defp get_errors(%{blueprint: blueprint}) when is_map(blueprint) do
+    case Map.get(blueprint, :result) do
+      result when is_map(result) -> Map.get(result, :errors)
+      _ -> nil
+    end
+  end
+
+  defp get_errors(_), do: nil
+
+  defp get_graphql_selections(%{blueprint: blueprint}) when is_map(blueprint) do
+    operation = current_operation(blueprint)
+
+    case operation do
+      nil ->
+        []
+
+      operation ->
+        case Map.get(operation, :selections) do
+          selections when is_list(selections) ->
+            selections
+            |> Enum.map(fn selection -> Map.get(selection, :name) end)
+            |> Enum.uniq()
+
+          _ ->
+            []
+        end
+    end
+  end
+
+  defp get_graphql_selections(_), do: []
+
+  defp get_operation_type(%{blueprint: blueprint}) when is_map(blueprint) do
+    operation = current_operation(blueprint)
+
+    case operation do
+      nil -> nil
+      operation -> Map.get(operation, :type)
+    end
+  end
+
+  defp get_operation_type(_), do: nil
+
+  defp get_operation_name(%{blueprint: blueprint}) when is_map(blueprint) do
+    operation = current_operation(blueprint)
+
+    case operation do
+      nil -> nil
+      operation -> Map.get(operation, :name)
+    end
```

Comment:
This pattern is used a few times. Could be simpler to use `get_in/2`, as long as it's not using structs because it's `nil` safe:

```suggestion
    blueprint |> current_operation() |> get_in([:name])
```

Reply:
Yeah, I keep getting the `Access behaviour` error when I use get_in with most of the metadata coming back from Absinthe.